README.adoc

```diff
@@ -1,34 +1,125 @@
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.hazelcast</groupId>
+    <artifactId>hazelcast-all</artifactId>
+    <version>${version.hazelcast}</version>
+</dependency>
+----
+
+Hazelcast configuration (`hazelcast.xml`) is placed under `src/main/resources/` folder. Now, we only need to auto-wire `HazelcastInstance` bean in `CommandController` and use it to access to Hazelcast data structures:
+
+[source, java]
+----
+package guides.hazelcast.springboot;
+
+import com.hazelcast.core.HazelcastInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.concurrent.ConcurrentMap;
+
+@RestController
+public class CommandController {
+    @Autowired
+    HazelcastInstance hazelcastInstance;
+
+    private ConcurrentMap<String,String> retrieveMap() {
+        return hazelcastInstance.getMap("map");
+    }
+
+    @RequestMapping("/put")
+    public CommandResponse put(@RequestParam(value = "key") String key, @RequestParam(value = "value") String value) {
+        retrieveMap().put(key, value);
+        return new CommandResponse(value);
+    }
+
+    @RequestMapping("/get")
+    public CommandResponse get(@RequestParam(value = "key") String key) {
+        String value = retrieveMap().get(key);
+        return new CommandResponse(value);
+    }
+}
+----
+
+== Running the Sample Application
+
+Run the application using Maven on a console window:
+
+----
+mvn spring-boot:run -Dserver.port=8080
+----
+
+Then run the application again on another console window. But this time, note that you need to set a different value for the `server.port`.
+
+----
+mvn spring-boot:run -Dserver.port=8081
+----
+
+After both application instances initialized, you should see the Hazelcast cluster formation in the output similar below:
+
+----
+2020-04-03 19:41:20.846  INFO 14501 --- [.IO.thread-in-0] c.h.internal.nio.tcp.TcpIpConnection     : [192.168.1.64]:5702 [hazelcast-cluster] [4.0] Initialized new cluster connection between /192.168.1.64:63783 and /192.168.1.64:5701
+2020-04-03 19:41:26.856  INFO 14501 --- [ration.thread-0] c.h.internal.cluster.ClusterService      : [192.168.1.64]:5702 [hazelcast-cluster] [4.0]
+
+Members {size:2, ver:2} [
+	Member [192.168.1.64]:5701 - 520aec3f-58a6-4fcb-a3c7-498dcf37d8ff
+	Member [192.168.1.64]:5702 - 5c03e467-d457-4847-b49a-745a335db557 this
+]
+
+2020-04-03 19:41:27.864  INFO 14501 --- [           main] com.hazelcast.core.LifecycleService      : [192.168.1.64]:5702 [hazelcast-cluster] [4.0] [192.168.1.64]:5702 is STARTED
+----
+
+Now, you can make calls to application endpoints to put data and get it back. Firstly, run the following command to put the data into Hazelcast distributed map:
+
+----
+curl "localhost:8080/put?key=key1&value=hazelcast"
```

Comment:
Check this curl commands. I guess this one won't work, because of `&` sign which is not escaped.

Reply:
No, they work, I double checked.